From b56f141dabfaa9587cf89b1f880895dbd3c478fe Mon Sep 17 00:00:00 2001
From: Matthew Schwartz <matthew.schwartz@linux.dev>
Date: Sat, 20 Dec 2025 19:49:24 -0800
Subject: [PATCH 051/176] [FROM-ML] mmc: rtsx_pci_sdmmc: increase delay after
 power on for SD cards

An mdelay of 1 leads to unreliable SD card initialization on some cards,
such as SanDisk Extreme PRO UHS-I microSD models, where they will be in a
"card stuck busy" state after power on. Increasing the delay to 10 ms
improves reliability from a 20% failure rate to 0% on two devices with
RTS525A card readers after testing 30 s2idle/resume cycles.

Signed-off-by: Matthew Schwartz <matthew.schwartz@linux.dev>
---
 drivers/mmc/host/rtsx_pci_sdmmc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/rtsx_pci_sdmmc.c b/drivers/mmc/host/rtsx_pci_sdmmc.c
index ab7dbc7c7a18..dd3c0f2cad38 100644
--- a/drivers/mmc/host/rtsx_pci_sdmmc.c
+++ b/drivers/mmc/host/rtsx_pci_sdmmc.c
@@ -937,7 +937,7 @@ static int sd_power_on(struct realtek_pci_sdmmc *host, unsigned char power_mode)
 	if (err < 0)
 		return err;
 
-	mdelay(1);
+	mdelay(10);
 
 	err = rtsx_pci_write_register(pcr, CARD_OE, SD_OUTPUT_EN, SD_OUTPUT_EN);
 	if (err < 0)
-- 
2.55.0

